File: projects/order-creation-queries/core/src/main/java/com/robinfood/core/mappers/BrandMappers.java
package com.robinfood.core.mappers;

import com.robinfood.core.dtos.BrandDTO;
import com.robinfood.core.dtos.configuration.StoreMultiBrandDTO;
import com.robinfood.core.dtos.menu.MenuBrandDTO;

public final class BrandMappers {

    private BrandMappers() {
        throw new IllegalStateException("Utility class");
    }

    public static BrandDTO menuBrandDtoToBrandDTO(MenuBrandDTO menuBrandDTO) {
        return BrandDTO.builder()
                .color(menuBrandDTO.getColor())
                .image(menuBrandDTO.getImage())
                .name(menuBrandDTO.getName())
                .build();
    }

    public static BrandDTO multiBrandDtoToBrandDTO(StoreMultiBrandDTO storeMultiBrandDTO) {
        return BrandDTO.builder()
                .color(storeMultiBrandDTO.getColor())
                .image(storeMultiBrandDTO.getImage())
                .name(storeMultiBrandDTO.getName())
                .build();
    }
}

File: projects/posv2/core/src/main/kotlin/com/robinfood/core/mappers/BrandMappers.kt
package com.robinfood.core.mappers

import com.robinfood.core.dtos.transactionrequest.BrandDTO
import com.robinfood.core.entities.transactionrequest.BrandEntity

fun BrandDTO.toBrandEntity(): BrandEntity {
    return BrandEntity(
        id = id,
        name = name
    )
}

fun BrandEntity.toBrandDTO(): BrandDTO {
    return BrandDTO(
            id = id,
            name = name
    )
}